learn/tutorials/adding-typescript-views-to-a-vaadin-14-app/content.adoc

```diff
@@ -0,0 +1,429 @@
+= Adding TypeScript Views to a Vaadin 14 App
+
+:type: text
+:tags: migration
+:description: We migrate a Vaadin 14 application to Vaadin 15, and then add a client-side view in TypeScript
+:repo:
+:linkattrs:
+:imagesdir: ./images
+:related_tutorials:
+
+Support for TypeScript views is one of the new features in Vaadin platform 15. TypeScript views have advantages of JavaScript frontend, such as faster loading, smaller UI response latency, and offline support, while still having static type checking similar to Java.
+
+In this tutorial we migrate a Vaadin 14 application to Vaadin 15, and then add a client-side view in TypeScript.
+
+== Prerequisites
+ 
+The steps in this tutorial start from an existing Vaadin 14 application. For example, you can take link:https://vaadin.com/start/lts/simple-ui[Simple App] starter.
+
+As usual for developing Vaadin 14 applications, you need Java, Maven, Node.js, and npm installed. We recommend to use an IDE with Java and TypeScript support.
+
+== Upgrade Vaadin dependency version
+
+Edit the `pom.xml` file and change the version of Vaadin platform dependency to Vaadin 15, as follows:
+
+.pom.xml excerpt with the Vaadin dependency declaration
+[source, xml]
+----
+<properties>
+
+    ...
+    <!-- Find the latest version on https://github.com/vaadin/platform/releases -->
+    <vaadin.version>15.0.0.alpha7</vaadin.version>
+    ...
+
+</properties>
+
+...
+
+<dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-bom</artifactId>
+            <version>${vaadin.version}</version>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+----
+
+Run the Maven `install` goal to verify the changes above are correct, e. g., using the terminal command:
+
+[source, bash]
+----
+$ mvn install
+----
+
+== Add new frontend dependencies
+
+We will use link:https://vaadin.com/router[Vaadin Router] for client-side views routing, and also link:https://lit-element.polymer-project.org[lit-element] for creating TypeScript views.
+
+Run the following command to install these and add them to `package.json`:
+
+[source, bash]
+----
+$ npm install --save @vaadin/router lit-element
+----
+
+== Create client-side entrypoints
+
+Create the `frontend/index.html` and `frontend/index.ts` files.
```

Comment:
The instructions output to the console instructs the user to copy the files from the target folder. Should we use the same approach here?

Reply:
Done.